src/core/ad.jl

```diff
@@ -287,7 +287,12 @@ end
 
 # Tracker's implementation of ldiv isn't good. We'll use Zygote's instead.
 const TrackedVecOrMat = Union{Tracker.TrackedVector, Tracker.TrackedMatrix}
-zygote_ldiv(A::AbstractMatrix, B::AbstractVecOrMat) = A \ B
+function zygote_ldiv(A::AbstractMatrix, B::AbstractVecOrMat)
```

Comment:
@mohamed82008 could you please explain why this solves the perf issues?

Reply:
This is a side issue that @yebai noticed. `TuringMvNormal` was slower than `MvNormal`, double more so for `ForwardDiff.Dual`. The reason can be found here https://github.com/JuliaStats/PDMats.jl/issues/99. `A \ B` seems to copy the matrix `A` first for some reason not clear to me. This causes a slowdown, even more so when the elements of `A` need to be converted to `Dual` in the copy. This change here avoids that.

Reply:
It was slower than `MvNormal` with the latest release of `PDMats` but `MvNormal` itself got slower in `PDMats` master which is why I opened that issue. It seems that `c::Cholesky \ x::Vector` used the in-place version which made it faster than `c.U' \ x`, which made me raise an eyebrow!

Reply:
I'm confused as to why `TuringMvNormal` was ever getting called with `ForwardDiff.Dual`s -- `TuringMvNormal` should only be used with `Tracker` when we're doing reverse-mode. Is there a good reason that this should ever be getting called?

Reply:
Not really! It's just nice to be fast anyways!

Reply:
If somebody decides to use `TuringMvNormal` directly for whatever reason, and then use `forward_diff` as the backend this will be useful. That's what @yebai did. But I know this is not the intended use of `TuringMvNormal`.

Reply:
Fair enough :)